scripts/prove/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+use std::{
+    fs,
+    time::{Duration, Instant},
+};
+
+use anyhow::{Ok, Result};
+use kona_host::HostCli;
+use op_succinct_host_utils::{
+    fetcher::{BlockInfo, OPSuccinctDataFetcher},
+    witnessgen::WitnessGenExecutor,
+};
+use sp1_sdk::{ExecutionReport, ProverClient, SP1Stdin};
+
+pub const DEFAULT_RANGE: u64 = 5;
+
+pub const MULTI_BLOCK_ELF: &[u8] = include_bytes!("../../../elf/range-elf");
+
+pub async fn generate_witness(host_cli: &HostCli) -> Result<Duration> {
+    let start_time = Instant::now();
+
+    // Start the server and native client.
+    let mut witnessgen_executor = WitnessGenExecutor::default();
+    witnessgen_executor.spawn_witnessgen(host_cli).await?;
+    witnessgen_executor.flush().await?;
+
+    let witness_generation_time_sec = start_time.elapsed();
+    println!(
+        "Witness Generation Duration: {:?}",
+        witness_generation_time_sec.as_secs()
+    );
+
+    Ok(witness_generation_time_sec)
+}
+
+pub async fn execute_multi(
+    prover: &ProverClient,
+    data_fetcher: &OPSuccinctDataFetcher,
+    sp1_stdin: SP1Stdin,
+    l2_chain_id: u64,
+    l2_start_block: u64,
+    l2_end_block: u64,
+) -> Result<(Vec<BlockInfo>, ExecutionReport, Duration)> {
+    let start_time = Instant::now();
+    let (_, report) = prover
+        .execute(MULTI_BLOCK_ELF, sp1_stdin.clone())
+        .run()
+        .unwrap();
+    let execution_duration = start_time.elapsed();
+
+    // Create the report directory if it doesn't exist.
+    let report_dir = format!("execution-reports/multi/{}", l2_chain_id);
+    if !std::path::Path::new(&report_dir).exists() {
+        fs::create_dir_all(&report_dir)?;
+    }
```

Comment:
Taste, but this shouldn't occur inside of this function, should instead be inside of the binary/test that's running it, as you materialize the `report_path` separately there as well. Would remove this and move it to the `multi` script.